### ISSUE TITLE
More detail on typed routing

### ISSUE THREAD

**Issue by drdamour** [typesafehub/activator-hello-play#14]
New to play (and all of typesafe including scala) and found this part rather confusing:

By default the routes are compiled by Play so that they can be used in a type-safe way. This prevents issues caused by incorrect URL usages.

This seems not-insignificant, an opinion i found supported by http://stackoverflow.com/questions/15697453/convention-over-configuration-in-play-2-x .

I thought about adding an example there were the user changes index to indexDNE and tries to compile and sees that it fails to compile.  I'm worried that wouldn't cover the intent of the statement.  Any thoughts? happy to submit a pull request tweak the tutorial if it makes sense.

**Comment by jamesward** [comment]
There are few ways to illustrate the typed routing.  One is the example you give where you have something wrong in the `routes` file.  It could be a syntax issue like misspelling an HTTP verb or not following the right way to do paths.  Also if you point to a controller method that doesn't exist that would also cause a compile error.

The other side of typed routing is that reverse routing is also typed.  So if you want to do a redirect or reference a URL instead of hard coding a string like:

```
def foo = Action(Redirect("/bar"))
```

The type-safe way to do that is with the reverse routing table (which is compiled from the `routes` file):

```
def foo = Action(Redirect(routes.Application.bar))
```

That way if the URL is changed in the routes file you won't need to make sure you manually update all of the references to URL.

Let me know if that helps.

**Comment by drdamour** [comment]
ok cool.  Seems like this would be the right tutorial to put this info in,

My first thought is it should be a new section in this tutorial explaining the different features of typed routing, probably after the view templates section so it can show reverse routing checks by example.

My second thought is that the examples could also be sprinkled about in various sections:
- bad verb in routing section
- bad method in controllers section
- bad reverse route in views section

any opinions how i should do it?

**Comment by jamesward** [comment]
Yeah, the stuff about the `routes` file should go in the HTTP Routing section.  The stuff about the reverse routing can be sprinkled into the Controller & Views sections.

Thanks!
